soh/soh/Enhancements/randomizer/3drando/location_access/locacc_ganons_castle.cpp

```diff
@@ -376,4 +369,54 @@ void RegionTable_Init_GanonsCastle() {
                   EventAccess(&logic->LightTrialClear, {[]{return logic->CanUse(RG_LIGHT_ARROWS);}}),
   }, {}, {});
   }
+
+  /*--------------------------
+  |     TOWER AND ESCAPE     |
+  ---------------------------*/
+  areaTable[RR_GANONS_TOWER_FLOOR_1] = Region("Ganon's Tower Floor 1", "Ganons Castle", {RA_GANONS_CASTLE}, NO_DAY_NIGHT_CYCLE, {}, {}, {
+                  //Exits
+                  Entrance(RR_GANONS_CASTLE_LOBBY,  {[]{return true;}}),
+                  Entrance(RR_GANONS_TOWER_FLOOR_2, {[]{return Here(RR_GANONS_TOWER_FLOOR_1, []{return logic->CanKillEnemy(RE_DINOLFOS, ED_CLOSE, true, 2);});}}),
+  });
+
+  areaTable[RR_GANONS_TOWER_FLOOR_2] = Region("Ganon's Tower Floor 2", "Ganons Castle", {RA_GANONS_CASTLE}, NO_DAY_NIGHT_CYCLE, {}, {
+                  //Locations
+                  LOCATION(RC_GANONS_TOWER_BOSS_KEY_CHEST, logic->CanKillEnemy(RE_STALFOS, ED_CLOSE, true, 2)),
+  }, {
+                  //Exits
+                  Entrance(RR_GANONS_TOWER_FLOOR_1, {[]{return true;}}),
+                  Entrance(RR_GANONS_TOWER_FLOOR_3, {[]{return Here(RR_GANONS_TOWER_FLOOR_2, []{return logic->CanKillEnemy(RE_STALFOS, ED_CLOSE, true, 2);});}}),
+  });
+
+  areaTable[RR_GANONS_TOWER_FLOOR_3] = Region("Ganon's Tower Floor 3", "Ganons Castle", {RA_GANONS_CASTLE}, NO_DAY_NIGHT_CYCLE, {}, {}, {
+                  //Exits
+                  Entrance(RR_GANONS_TOWER_FLOOR_2,        {[]{return true;}}),
+                  Entrance(RR_GANONS_TOWER_GANONDORF_LAIR, {[]{return Here(RR_GANONS_TOWER_FLOOR_3, []{return logic->CanKillEnemy(RE_IRON_KNUCKLE, ED_CLOSE, true, 2);}) && logic->HasItem(RG_GANONS_CASTLE_BOSS_KEY);}}),
+  });
+
+  areaTable[RR_GANONS_TOWER_GANONDORF_LAIR] = Region("Ganondorf's Lair", "Ganons Castle", {RA_GANONS_CASTLE}, NO_DAY_NIGHT_CYCLE, {}, {
+                  //Locations
+                  LOCATION(RC_GANONDORF_HINT, logic->HasBossSoul(RG_GANON_SOUL)),
+                  //18 pots
+  }, {
+                  //Exits
+                  Entrance(RR_GANONS_TOWER_FLOOR_3, {[]{return true;}}),
```

Comment:
If you warped into the ganondorf fight, you wouldn't be able to leave, so this exit shouldn't exist unless there's some special edge case I am unaware of.

Reply:
This area currently encompasses everything after the first boss door so in that case it should be split, don't know what to name the one in between the boss doors tho.

Reply:
OK I reviewed the area, and you don't seem able to go backwards through the boss key locked doors at all, which makes the whole point moot. There may be value in separating pot room from the boss if a void warp is found, but that doesn't sound like anything that's needed now